### PR TITLE
Migrate Sentry Android to v4.3.0

### PR DESCRIPTION
**Pull request by msasikanth**
https://app.clubhouse.io/simpledotorg/story/3251/migrate-sentry-android-to-v4-3-0

### CODE REVIEW

**Comment by msasikanth** [comment]
Sentry migration guide: https://docs.sentry.io/platforms/android/migration/

File: gradle.properties [hunk 1]
```diff
@@ -61,3 +61,21 @@ securityFallbackApiEndpoint=do_not_change_here
...
 # strips debug symbols from the APK.
 androidNdkVersion=21.3.6528147
+
+# Sentry Config
+## DSN
+qaSentryDsn=do_not_change_here
+sandboxSentryDsn=do_not_change_here
+stagingSentryDsn=do_not_change_here
+productionSentryDsn=do_not_change_here
+securitySentryDsn=do_not_change_here
```
Comment (vinaysshenoy) [inline gradle.properties:71]: The DSN is also same for all the builds. We only need one.

File: app/src/main/java/org/simple/clinic/crash/SentryCrashReporter.kt [hunk 2]
```diff
@@ -30,7 +31,11 @@ class SentryCrashReporter @Inject constructor(
 ) : CrashReporter {
 
   override fun init(appContext: Application) {
-    Sentry.init(AndroidSentryClientFactory(appContext))
+    Sentry.init { options ->
+      options.dsn = SENTRY_DSN
+      options.sampleRate = SENTRY_SAMPLE_RATE
+      options.environment = SENTRY_ENVIRONMENT
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/crash/SentryCrashReporter.kt:37]: The `sentry.properties` file also has an auth token which should be a secret. Where is that getting passed?

File: app/src/main/java/org/simple/clinic/crash/SentryCrashReporter.kt [hunk 3]
```diff
@@ -64,19 +69,19 @@ class SentryCrashReporter @Inject constructor(
...
-        .setLevel(priorityToLevel(breadcrumb.priority))
-        .setCategory(breadcrumb.tag)
-        .setMessage(breadcrumb.message)
-        .setTimestamp(Date(System.currentTimeMillis()))
-        .build()
-    Sentry.getContext().recordBreadcrumb(sentryBreadcrumb)
+    val sentryBreadCrumb = SentryBreadcrumb().apply {
+      level = priorityToLevel(breadcrumb.priority)
+      category = breadcrumb.tag
+      message = breadcrumb.message
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/crash/SentryCrashReporter.kt:81]: Is the timestamp set automatically by Sentry now?

**Review by vinaysshenoy — CHANGES_REQUESTED** [review]
Overall PR looks good, left a couple of comments.

File: app/src/main/java/org/simple/clinic/crash/SentryCrashReporter.kt (hunk 3 above)
Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/crash/SentryCrashReporter.kt:81]: Yep

File: app/src/main/java/org/simple/clinic/crash/SentryCrashReporter.kt (hunk 2 above)
Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/crash/SentryCrashReporter.kt:37]: Do we really need it? The API doesn't expose a way to set that. As far as I can tell from the docs, I don't find it as needed for the Android implementation?

File: gradle.properties (hunk 1 above)
Comment (msasikanth) [reply gradle.properties:71]: Oh okay, will change it

File: app/src/main/java/org/simple/clinic/crash/SentryCrashReporter.kt (hunk 2 above)
Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/crash/SentryCrashReporter.kt:37]: Found the use case for it Gradle plugin. Adding it there
https://docs.sentry.io/platforms/android/proguard/

**Review by vinaysshenoy — APPROVED** [review]

**Review by vinaysshenoy — APPROVED** [review]

**Pull request merged**